engineapi/snapshot.go

```diff
@@ -78,15 +79,18 @@ func (e *Engine) SnapshotPurge() error {
 	return nil
 }
 
-func (e *Engine) SnapshotBackup(snapName, backupTarget string, labels map[string]string) error {
+func (e *Engine) SnapshotBackup(snapName, backupTarget string, labels map[string]string, secret map[string]string) error {
 	snap, err := e.SnapshotGet(snapName)
 	if err != nil {
 		return errors.Wrapf(err, "error getting snapshot '%s', volume '%s'", snapName, e.name)
 	}
 	if snap == nil {
 		return errors.Errorf("could not find snapshot '%s' to backup, volume '%s'", snapName, e.name)
 	}
-	args := []string{"--url", e.cURL, "backup", "create", "--dest", backupTarget}
+	if strings.Contains(backupTarget, "s3") && (secret == nil || secret[types.SecretAccessKey] == "" || secret[types.SecretKey] == "") {
+		return errors.Errorf("could not backup snapshot '%s' without aws credential", snapName)
+	}
+	args := []string{"--url", e.cURL, "backup", "--accessKey", secret[types.SecretAccessKey], "--secretKey", secret[types.SecretKey], "create", "--dest", backupTarget}
```

Comment:
I don't think we need new options `--accessKey` and `--secretKey` for the engine. We can set them in the environment variable before executing the command, and Longhorn Engine can pick them up explicitly.

The problems with setting them in the cmdline are:
1. if the command failed, the log will spit out the credential in plaintext. Though it's not a deal-breaker.
2. Every time we are adding a new type of destination, we will add some more parameters to the engine in order to support it. I definitely prefer the credential to be passed implicitly instead.